dali/base/dasds.cpp

```diff
@@ -1407,17 +1427,36 @@ class CDeltaWriter : implements IThreaded
         CriticalBlock b(pendingCrit);
         addToQueue(item);
     }
-    void clear()
+    void flush()
     {
-        CriticalBlock b(pendingCrit);
-        pending = {};
+        bool waitOnEmptySem = false;
+        {
+            CriticalBlock b(pendingCrit);
+            if (waiting)
+            {
+                if (pendingSz)
+                    signalWaiting();
```

Comment:
Shouldn't this also wait until until signalEmpty occurs?  Othewise it might still be writing data (or not yet have written the data) before the function continues.

Reply:
it will because sianglWaiting resets waiting to false (all within pendingCrit), and so it will fall into the if (!waiting) block below.